deploy/crds/kieapp.crd.yaml

```diff
@@ -1471,6 +1468,9 @@ spec:
                         keystoreSecret:
                           description: Keystore secret name
                           type: string
+                        kieServerContainerDeployment:
+                          description: The Maven GAV to deploy, e.g., rhpam-kieserver-library=org.openshift.quickstarts:rhpam-kieserver-library:1.5.0-SNAPSHOT
+                          type: string
```

Comment:
@tchughesiv is this api change safe from point of auto updates? `kieServerContainerDeployment` is moved from server.Build object to server. I'm little worried that this change could broke down existing KieApps during updates.

@Kaitou786 As the issue is that s2i build is execute when not needed (and git properties are required), maybe it could be enough to not require git properties (make them optional, remove 'require' flag from api) and do not trigger S2I build when there is set only container GAV.

Reply:
correct @jakubschwan ... this change would break things and cannot be done without an api version increase. this would complicate the change significantly as upgrade logic would have to be added in the form of a conversion webhook.

only api additions are allowed without versioning the api.